### ISSUE TITLE
Safari&Performance: When scrolling, there is white background for a short time

### ISSUE THREAD

**Issue by mato75** [kamilkp/angular-vs-repeat#49]
When scrolling in Safari, there is a performance issue. When scrolling, depending on direction, there is a white background for a short time.

You can see it in the demo page.

**Comment by kamilkp** [comment]
This can be overcome by adjusting the `vs-excess` properly

**Comment by mato75** [comment]
vs-excess is set to 200, and still happens. Sometimes the whole list is not visible.

**Comment by kamilkp** [comment]
Can you create a jsfiddle?

**Comment by mato75** [comment]
http://kamilkp.github.io/angular-vs-repeat/#?tab=1

scroll fast, on you will see a white background, it is not that obvious here, but if you have a higher view, it is much easier to see.

**Comment by mato75** [comment]
http://jsbin.com/mobipezehi/1/edit?html,css,js,output - in this example you will see, when scrolling, white flashes

**Comment by mato75** [comment]
Any update on this issue?

**Comment by mato75** [comment]
using  window.requestAnimationFrame speeds up scrolling a bit, but it is still not good in safari and IE.

**Comment by Avi-mcqueen** [comment]
Join to mato75, 
Also have this issue, 

<div class="items-list" id="tableSearchResults" vs-repeat="85" vs-excess="10">
        <div class="request-row" ng-repeat="item in itemsList" ng-click="showItemDetails(request)">
        </div>
</div>

testing on Android device, render speed on medium-high speed scrolling is slow.
on desktop browser works good without this issue

**Comment by mato75** [comment]
Any update?

**Comment by SuperCoolSEBoy** [comment]
I am having the same experience. Unlike what it has on the demo page, which has a delay when you drag the scrollbar dragger with 1000 items, I can use the scrollbar dragger smoothly but the list flashes somehow. I am not quite sure but I guess this is because IE is slow in re-rendering the ng-repeat items (and of coz, Chrome is always better than IE regarding perf)?

**Comment by darrinholst** [comment]
ng-cells has amazing scroll performance. How do we get vs-repeat to be that responsive?

**Comment by darrinholst** [comment]
The degraded scrolling performance that I was seeing was due to adding `filter` and `orderBy` filters. That brought the digest cycle up to 150ms with just `orderBy` and 500ms with both activated. Need to figure out how to add sorting and filtering without killing scroll performance if anyone has any suggestions.

**Comment by Avi-mcqueen** [comment]
Made video example of my problem.
https://youtu.be/aG6W5mlMqxM

Anyone has an idea how to fix this?

**Comment by graphefruit** [comment]
+1 Here, experience this on WP8.1 aswell as on android.

Is there an update to this problem?

**Comment by samk-ayyagari** [comment]
@darrinholst @SuperCoolSEBoy @dudesweet100  Guys can please let me know how you have solved this problem.

**Comment by darrinholst** [comment]
I don't think we ever totally eliminated the white flash (I'm not on the project anymore), but it is very important to do all your formatting/filtering/sorting of data up front in the controller. You want to get that digest cycle time as small as possible. It's going to be very hard to match the scrolling performance of a native table so you have take some tradeoffs unfortunately.

**Comment by vTodorov** [comment]
+1
